internal/core/component_factory.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+#![warn(missing_docs)]
+
+//! This module contains the `ComponentFactory` and a `Component`. Both are
+//! exposed to the C API.
+
+use core::ffi::c_void;
+use core::fmt::Debug;
+
+use alloc::rc::Rc;
+
+use crate::{
+    api::ComponentHandle,
+    component::{ComponentRc, ComponentVTable},
+};
+
+/// A `Component` built by a `ComponentFactory`
```

Comment:
The documentation should say that it could be created from any type that implements the ComponentHandle trait